awx/resource_inventory_source.go

```diff
@@ -171,19 +178,21 @@ func resourceInventorySourceUpdate(ctx context.Context, d *schema.ResourceData,
 	}
 
 	updateInventorySourceData := map[string]interface{}{
-		"name":                 d.Get("name").(string),
-		"description":          d.Get("description").(string),
-		"enabled_var":          d.Get("enabled_var").(string),
-		"enabled_value":        d.Get("enabled_value").(string),
-		"overwrite":            d.Get("overwrite").(bool),
-		"overwrite_vars":       d.Get("overwrite_vars").(bool),
-		"update_on_launch":     d.Get("update_on_launch").(bool),
-		"inventory":            d.Get("inventory_id").(int),
-		"source":               d.Get("source").(string),
-		"source_vars":          d.Get("source_vars").(string),
-		"host_filter":          d.Get("host_filter").(string),
-		"update_cache_timeout": d.Get("update_cache_timeout").(int),
-		"verbosity":            d.Get("verbosity").(int),
+		"name":                     d.Get("name").(string),
+		"description":              d.Get("description").(string),
+		"enabled_var":              d.Get("enabled_var").(string),
+		"enabled_value":            d.Get("enabled_value").(string),
+		"overwrite":                d.Get("overwrite").(bool),
+		"overwrite_vars":           d.Get("overwrite_vars").(bool),
+		"update_on_launch":         d.Get("update_on_launch").(bool),
+		"inventory":                d.Get("inventory_id").(int),
+		"credential":               d.Get("credential_id").(int),
+		"source":                   d.Get("source").(string),
+		"source_vars":              d.Get("source_vars").(string),
+		"host_filter":              d.Get("host_filter").(string),
+		"update_cache_timeout":     d.Get("update_cache_timeout").(int),
+		"verbosity":                d.Get("verbosity").(int),
+		"execution_environment_id": d.Get("execution_environment_id").(int),
```

Comment:
From what I see in the doc here:
https://docs.ansible.com/automation-controller/4.4/html/controllerapi/api_ref.html#/Inventory_Sources/Inventory_Sources_inventory_sources_update_0
this field is named "execution_environment" in the PUT body, not "execution_environment_id".